### PR TITLE
BUG/PERF: MaskedArray.searchsorted(np.nan)

### PR DESCRIPTION
**Pull request by jbrockmendel**
Needed for follow-up to ExtensionIndex to avoid object-dtype cast for engine.

```
arr = pd.array(range(10**5))

%timeit arr.searchsorted(4)
2.87 ms ± 223 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
6.67 µs ± 178 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

### CODE REVIEW

File: pandas/core/arrays/masked.py [hunk 1]
```diff
@@ -681,6 +686,23 @@ def copy(self: BaseMaskedArrayT) -> BaseMaskedArrayT:
...
+        side: Literal["left", "right"] = "left",
+        sorter: NumpySorter = None,
+    ) -> npt.NDArray[np.intp] | np.intp:
+        if self._hasna:
+            raise ValueError(
+                "searchsorted requires array to be sorted, which is impossible "
+                "with NAs present."
+            )
+        if isinstance(value, ExtensionArray):
+            value = value.astype(object)
```
Comment (jreback) [inline pandas/core/arrays/masked.py:702]: why can't you *just* use the data here instead of casting?

Comment (jbrockmendel) [reply pandas/core/arrays/masked.py:702]: This is the value, not 'self'.  The base class does this casting too.

Comment (jreback) [reply pandas/core/arrays/masked.py:702]: ahh got it, then this is fine

**Comment by jreback** [comment]
>  # The base class casts to object dtype, for which searchsorted returns
        #  0 from the left and 10 from the right.

i think needs a whatsnew? as this is the bug right

**Pull request merged**